libhdf5/hdf5file.c

```diff
@@ -427,6 +436,11 @@ NC4_set_fill(int ncid, int fillmode, int *old_modep)
 
    nc4_info->fill_mode = fillmode;
 
+   /* set/change fill mode for all variables */
+   for (i=0; i<ncindexcount(grp->vars); i++) {
+       NC_VAR_INFO_T *var = (NC_VAR_INFO_T*) ncindexith(grp->vars, i);
```

Comment:
Need to handle case of var == NULL. I think this can occur when a variable is deleted - not sure, but we do handle it everywhere else with

if (!var) continue;

Reply:
I guess when var == NULL it means no variable exits and thus `ncindexcount(grp->vars)` should return zero and the loop is skipped. Correct?